pandas/tseries/period.py

```diff
@@ -308,13 +280,18 @@ def _simple_new(cls, values, name=None, freq=None, **kwargs):
             else:
                 return cls(values, name=name, freq=freq, **kwargs)
 
+        return cls._from_ordinals(values, name, freq, **kwargs)
+
+    @classmethod
+    def _from_ordinals(cls, values, name=None, freq=None, **kwargs):
```

Comment:
I see that this is private, ok with that. which makes sense as user's don't need to know how the sausage is made.